go-controller/pkg/ovn/master.go

```diff
@@ -351,6 +355,52 @@ func (oc *Controller) deleteNode(nodeName string, nodeSubnet *net.IPNet) error {
 	return nil
 }
 
+// OVN uses an overlay and doesn't need GCE Routes, we need to
+// clear the NetworkUnavailable condition that kubelet adds to initial node
+// status when using GCE (done here: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L237).
+// See discussion surrounding this here: https://github.com/kubernetes/kubernetes/pull/34398.
+// TODO: make upstream kubelet more flexible with overlays and GCE so this
+// condition doesn't get added for network plugins that don't want it, and then
+// we can remove this function.
+func (oc *Controller) clearInitialNodeNetworkUnavailableCondition(origNode *kapi.Node) {
+	// Informer cache should not be mutated, so get a copy of the object
+	node := origNode.DeepCopy()
+	knode := node
+	cleared := false
+	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		var err error
+
+		if knode != node {
+			knode, err = oc.kube.GetNode(node.Name)
```

Comment:
careful, `knode` is a pointer to the shared cache - you have to `DeepCopy()` before mutating it.

(I don't understand the difference between `node` and `knode`, maybe you can clean that up?)